File: servlet-app/src/main/java/ru/deyeva/servletapp/controller/CatController.java
package ru.deyeva.servletapp.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.web.bind.annotation.*;
import ru.deyeva.servletapp.dto.CatDto;
import ru.deyeva.servletapp.entity.Cat;
import ru.deyeva.servletapp.repository.CatRepository;

import java.util.List;

@RequiredArgsConstructor
@RestController
@RequestMapping("/cat")
public class CatController {

    private final CatRepository catRepository;

    @PostMapping
    private void saveCat(@RequestBody CatDto catDto) {
        Cat cat = new Cat(null, catDto.name(), catDto.age(), catDto.gender(), catDto.color());
        catRepository.save(cat);
    }

    @GetMapping
    public List<Cat> getCatsByName(@RequestParam String name) {
        return catRepository.findAllByName(name);
    }
}

File: coroutine-app/src/main/kotlin/ru/deyeva/coroutineapp/controller/CatController.kt
package ru.deyeva.coroutineapp.controller

import kotlinx.coroutines.flow.Flow
import org.springframework.web.bind.annotation.*
import ru.deyeva.coroutineapp.dto.CatDto
import ru.deyeva.coroutineapp.repository.CatRepository

@RestController
@RequestMapping("/cat")
class CatController(
    val catRepository: CatRepository
) {
    @PostMapping
    suspend fun saveCat(@RequestBody cat: CatDto) {
        catRepository.save(cat.name, cat.age, cat.gender, cat.color)
    }

    @GetMapping
    suspend fun getCatsByName(@RequestParam name: String): Flow<Map<String, Any?>> {
        return catRepository.findAllByName(name)
    }
}